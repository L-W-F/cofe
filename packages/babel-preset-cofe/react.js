module.exports = () => ({
  presets: ['./typescript', '@babel/preset-react'].map(require.resolve),
});
