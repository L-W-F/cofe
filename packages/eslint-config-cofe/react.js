module.exports = {
  extends: [
    './typescript',
    './rules/jsx-a11y',
    './rules/react',
    './rules/react-hooks',
  ].map(require.resolve),
  rules: {},
};
