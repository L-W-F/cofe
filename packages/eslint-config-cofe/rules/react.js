module.exports = {
  plugins: ['react'],

  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },

  settings: {
    react: { version: 'detect' },
  },

  rules: {
    // https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules
    'react/forbid-foreign-prop-types': ['error', { allowInPropTypes: true }],
    'react/jsx-filename-extension': ['error', { extensions: ['.jsx', '.tsx'] }],
    'react/jsx-indent': ['error', 2],
    'react/jsx-indent-props': ['error', 2],
    'react/jsx-no-comment-textnodes': 'error',
    'react/jsx-no-duplicate-props': 'error',
    'react/jsx-no-target-blank': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-pascal-case': [
      'error',
      {
        allowAllCaps: true,
        ignore: [],
      },
    ],
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/no-danger-with-children': 'error',
    // Disabled because of undesirable warnings
    // See https://github.com/facebook/create-react-app/issues/5204 for
    // blockers until its re-enabled
    // 'react/no-deprecated': 'error',
    'react/no-direct-mutation-state': 'error',
    'react/no-is-mounted': 'error',
    'react/no-typos': 'error',
    'react/react-in-jsx-scope': 'off',
    'react/require-render-return': 'error',
    'react/self-closing-comp': [
      'error',
      {
        component: true,
        html: true,
      },
    ],
    'react/style-prop-object': 'error',
  },
};
