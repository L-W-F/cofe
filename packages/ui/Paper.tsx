import React from 'react';
import { Box, BoxProps } from '@chakra-ui/react';
import { withSx } from './withSx';

export interface PaperProps extends BoxProps {}

export const Paper = withSx<PaperProps>('Paper')(
  ({ rounded = 'md', shadow = 'xs', overflow = 'hidden', ...props }) => {
    return (
      <Box rounded={rounded} shadow={shadow} overflow={overflow} {...props} />
    );
  },
);
