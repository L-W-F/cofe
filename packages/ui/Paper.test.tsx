import React from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { create } from 'react-test-renderer';
import { Paper } from './Paper';

test('Paper', () => {
  const component = create(
    <ChakraProvider>
      <Paper>Paper</Paper>
    </ChakraProvider>,
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
