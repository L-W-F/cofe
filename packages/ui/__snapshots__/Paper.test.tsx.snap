// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper 1`] = `
Array [
  <div
    className="css-15gz6je"
  >
    Paper
  </div>,
  <span />,
]
`;
